### PR TITLE
use TRI_read_t where appropriate

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8148/

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8166/

File: lib/Basics/FileUtils.cpp [hunk 1]
```diff
@@ -184,7 +184,7 @@ static void fillStringBuffer(int fd, std::string const& filename,
     if (result.reserve(chunkSize) != TRI_ERROR_NO_ERROR) {
       THROW_ARANGO_EXCEPTION(TRI_ERROR_OUT_OF_MEMORY);
     }
-    ssize_t n = TRI_READ(fd, result.end(), static_cast<TRI_read_t>(chunkSize));
+    TRI_read_t n = TRI_READ(fd, result.end(), static_cast<TRI_read_t>(chunkSize));
```
Comment (jsteemann) [inline lib/Basics/FileUtils.cpp]: I don't think this makes sense.
The signature for `read` on Linux is:
```
ssize_t read(int fd, void *buf, size_t count);
``` 
So the parameter type for the function return type and the call argument are different.
This PR makes things worse. What is the motivation for this?

Comment (dothebart) [reply lib/Basics/FileUtils.cpp]: on linux `TRI_read_t` *is* `ssize_t`; However, ssize_t is not present on windows. 

as its name says, `TRI_read_t` is intended to be the type to be returned by `TRI_READ`.

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8421/

File: CMakeLists.txt [hunk 2]
```diff
@@ -517,11 +505,6 @@ endif()
 ## ARCHITECTURE
 ################################################################################
 
-if (SOLARIS OR NOT CMAKE_SIZEOF_VOID_P)
```
Comment (jsteemann) [inline CMakeLists.txt]: mmh, is this safe? This is `OR`-combined, not `AND`-combined...

File: lib/Basics/tri-zip.cpp [hunk 3]
```diff
@@ -207,7 +207,7 @@ static int ExtractCurrentFile(unzFile uf, void* buffer, size_t const bufferSize,
     }
 
     while (true) {
-      int result = unzReadCurrentFile(uf, buffer, (unsigned int)bufferSize);
+      TRI_read_reply_t result = unzReadCurrentFile(uf, buffer, (unsigned int)bufferSize);
```
Comment (jsteemann) [inline lib/Basics/tri-zip.cpp]: the return type of `unzReadCurrentFile` is always an `int`:
```
3rdParty/zlib/zlib-1.2.11/contrib/minizip/unzip.h:extern int ZEXPORT unzReadCurrentFile OF((unzFile file,
```

File: lib/Basics/tri-zip.cpp [hunk 4]
```diff
@@ -419,7 +419,7 @@ int TRI_Adler32(char const* filename, uint32_t& checksum) {
     return TRI_set_errno(TRI_ERROR_SYS_ERROR);
   }
 
-  ssize_t chunkRemain = static_cast<TRI_read_t>(statbuf.st_size);
+  TRI_read_t chunkRemain = static_cast<TRI_read_reply_t>(statbuf.st_size);
```
Comment (jsteemann) [inline lib/Basics/tri-zip.cpp]: this makes no sense, as we are first casting to `TRI_read_reply_t` and then implicitly cast to `TRI_read_t` :thinking:

File: lib/SimpleHttpClient/ClientConnection.cpp [hunk 5]
```diff
@@ -188,7 +185,7 @@ bool ClientConnection::readClientConnection(StringBuffer& stringBuffer, bool& co
       return true;
     }
 #ifdef ARANGODB_ENABLE_MAINTAINER_MODE
-    _read += (uint64_t)lenRead;
+    _read += static_cast<TRI_read_t>(lenRead);
```
Comment (jsteemann) [inline lib/SimpleHttpClient/ClientConnection.cpp]: `_read` is a `uint64_t`, so casting to something else is pointless
```suggestion
    _read += static_cast<uint64_t>(lenRead);
```

File: lib/V8/v8-shell.cpp [hunk 6]
```diff
@@ -217,7 +217,7 @@ static void JS_ProcessCsvFile(v8::FunctionCallbackInfo<v8::Value> const& args) {
   char buffer[10240];
 
   while (true) {
-    ssize_t n = TRI_READ(fd, buffer, sizeof(buffer));
+    TRI_read_reply_t n = TRI_READ(fd, buffer, sizeof(buffer));
```
Comment (jsteemann) [inline lib/V8/v8-shell.cpp]: It is inconsistent to not cast here.
```suggestion
    TRI_read_reply_t n = TRI_READ(fd, buffer, static_cast<TRI_read_t>(sizeof(buffer)));
```

File: arangosh/Utils/ManagedDirectory.cpp [hunk 7]
```diff
@@ -171,9 +171,9 @@ inline void rawWrite(int fd, char const* data, size_t length,
 }
 
 /// @brief Performs a raw (non-decrypted) read
-inline ssize_t rawRead(int fd, char* buffer, size_t length, arangodb::Result& status,
+inline TRI_read_reply_t rawRead(int fd, char* buffer, size_t length, arangodb::Result& status,
```
Comment (jsteemann) [inline arangosh/Utils/ManagedDirectory.cpp]: `TRI_read_reply_t` is a strange name. It should rather be `TRI_read_return_t`.

File: CMakeLists.txt (hunk 2 above)
Comment (dothebart) [reply CMakeLists.txt]: its most probably only relevant for crosscompiling.

File: lib/Basics/tri-zip.cpp [hunk 8]
```diff
@@ -419,7 +419,7 @@ int TRI_Adler32(char const* filename, uint32_t& checksum) {
     return TRI_set_errno(TRI_ERROR_SYS_ERROR);
   }
 
-  ssize_t chunkRemain = static_cast<TRI_read_t>(statbuf.st_size);
+  TRI_read_t chunkRemain = static_cast<TRI_read_return_t>(statbuf.st_size);
```
Comment (jsteemann) [inline lib/Basics/tri-zip.cpp]:
```suggestion
  TRI_read_t chunkRemain = static_cast<TRI_read_t>(statbuf.st_size);
```

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8440/

**Pull request merged**